### ISSUE TITLE
huggingface transformers 4.21.* and OnnxRuntime has different results on BERT/GPT-2

### ISSUE THREAD

**Issue by tianleiwu** [microsoft/onnxruntime#12843]
**Describe the bug**

Recently, we found that there is parity issue between OnnxRuntime and transformers 4.21 or newer version. It is because transformers recently change the attention mask from -1e4 to torch.finfo(dtype).min (see https://github.com/huggingface/transformers/pull/17306). However, the attention kernel in OnnxRuntime still uses -1e4 for attention mask. 

It impacts ONNX models with Attention operator (like BERT, Roberta and GPT-2). For example, distilgpt2 model in mixed precision has about 1% chance of different top 1 token per step with greedy search.

The following scenarios could be impacted potentially:
(1) Your model (BERT, Roberta and GPT-2 etc) is trained with transformers 4.21 or later, then inference with OnnxRuntime. You might notice accuracy loss.
(2) Your original model is fp32 and you convert it to fp16 for inference.  fp32 model and fp16 model will get different result since fp32 and fp16 minimal value is different. The impact might be insignificant since converting fp32 to fp16 itself has accuracy loss.

This thread tracks the issue and solution.

A candidate solution is to add an attribute to Attention operator to indicates the mask value.

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- ONNX Runtime installed from (source or binary):
- ONNX Runtime version:
- Python version:
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.
